Issue3538 air2 air recovery

SenHuang19 opened the pull request:
This is to address https://github.com/lbl-srg/modelica-buildings/pull/3554#pullrequestreview-2528487023

SenHuang19 commented:
@JayHuLBL can you please review this PR?

JayHuLBL commented:
@SenHuang19 
How did you address the comment: `The examples use sta_an and/or sta_b, but these are not defined at zero flow rate. No example should use these quantities. (We removed them in other models due to verification issues across tools.)`?

SenHuang19 commented:
> @SenHuang19 How did you address the comment: `The examples use sta_an and/or sta_b, but these are not defined at zero flow rate. No example should use these quantities. (We removed them in other models due to verification issues across tools.)`?

@JayHuLBL for some examples (e.g., <code>Buildings.Fluid.HeatExchangers.ThermalWheels.Latent.BaseClasses.Validation.HeatExchangerWithInputEffectiveness</code>), I replace the sta_a/sta_b related quantities with sensors (e.g., replacing <code>hex.sta_a1.T</code> with <code>senExhTemIn.T</code>).
For the rest examples (e.g., <code>Buildings.Fluid.HeatExchangers.ThermalWheels.Latent.Examples.SpaceCooling</code>), I deleted the sta_a/sta_b related quantities (e.g., <code>cooCoi.sta_a2.T</code>) as they are not relevant to the thermal wheel.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Latent/Examples/SpaceCooling.mo:
```diff
@@ -18,7 +18,13 @@ model SpaceCooling "Space cooling system"
+    mExh_flow_nominal=mA_flow_nominal,
+    dpSup_nominal=200,
+    haveVariableSpeed=false)
+    "Performance record for the sensible heat wheel"
```
Is it for the sensible heat wheel?

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Data/Generic.mo:
```diff
@@ -86,12 +119,12 @@ corrections versus wheel speed ratio are correction factors that are multiplied
 </li>
 <li>
-When <code>haveLatentHeatExchange = true</code>,
+When <code>haveLatentHeatExchange=true</code>,
```
Has the flag been used to enable/disable parameters? I didn't see it.

SenHuang19 replied:
no, it was not used. I fixed this issue.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Data/Generic.mo:
```diff
@@ -1,37 +1,71 @@
 within Buildings.Fluid.HeatExchangers.ThermalWheels.Data;
-record Generic "Generic data record for variable-speed wheels"
+record Generic "Generic data record for thermal wheels"
```
In the class `Buildings.Fluid.HeatExchangers.ThermalWheels.Data.ASHRAE`, please `final haveVariableSpeed=true,` as below. I believe the ASHRAE is only for the wheels with variable speed drive, right?
```Modelica
record ASHRAE = Buildings.Fluid.HeatExchangers.ThermalWheels.Data.Generic(
  final haveVariableSpeed=true,
  senHeatExchangeEffectiveness(
```
With the change, when instantiating it, like in `Buildings.Fluid.HeatExchangers.ThermalWheels.Latent.Validation.SpeedControlled`, uses will not be able to change the value.

SenHuang19 replied:
add it as suggested

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Sensible/BaseClasses/PartialWheel.mo:
```diff
@@ -5,34 +5,9 @@ partial model PartialWheel
-    "Part load (75% of the nominal supply flow rate) sensible heat exchanger effectiveness at the heating mode"
-    annotation (Dialog(group="Part load effectiveness"));
+  parameter Buildings.Fluid.HeatExchangers.ThermalWheels.Data.Generic per
+    "Record with performance data"
```
```Modelica
per(final haveLatentHeatExchange=false)
```

SenHuang19 replied:
add it as suggested

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Latent/BaseClasses/PartialWheel.mo:
```diff
@@ -5,50 +5,10 @@ partial model PartialWheel
-    "Part load (75% of the nominal supply mass flow rate) latent heat exchanger effectiveness at the heating mode"
-    annotation (Dialog(group="Part load effectiveness"));
+
+  parameter Buildings.Fluid.HeatExchangers.ThermalWheels.Data.Generic per
```
```Modelica
per(final haveLatentHeatExchange=true)
```

SenHuang19 replied:
add it as suggested

JayHuLBL commented on Buildings/Fluid/HeatExchangers/ThermalWheels/Latent/BaseClasses/PartialWheel.mo:
```diff
@@ -88,67 +48,44 @@ partial model PartialWheel
-    "Exhaust air temperature"
-    annotation (Placement(transformation(extent={{-160,-40},{-140,-20}})));
+
+initial equation
```
There is no need to have the `assert()` function, if we can finalize the flag `haveLatentHeatExchange`.

JayHuLBL reviewed (changes requested):
@SenHuang19  Please see the inline comments.
